### ISSUE TITLE
ag-grid installation doesn't work with React 18

### ISSUE THREAD

**Issue by juhahinkula** [ag-grid/ag-grid#5090]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
**Providing a Reproducible Scenario**
1. create new React project using the latest version of create-react-app (react-scripts 5.0.1. and React v 18)
2. install ag-grid: npm install --save ag-grid-community ag-grid-react

**Current behavior**
You get following error in ag-grid installation:

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: reacttest@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.3.0 || ^17.0.0" from ag-grid-react@27.1.0
npm ERR! node_modules/ag-grid-react
npm ERR!   ag-grid-react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps

**Expected behavior**
Installation should finish without errors

**Please tell us about your environment:**
Windows 10
VSCode

* **AG Grid version:** 
27.1.0
 
* **Browser:**
Chrome

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6

**Comment by MisterCrispyFL** [comment]
Just to add, if you force the install, it will throw these errors in the console:
ResizeObserver.observe: Argument 1 is not an object.
and
The above error occurred in the <RowContainerComp> component

**Comment by ag-grid-shared** [comment]
Regards ResizeObserver.observe, we are currently fixing this issue. It's to do with React 18's Reusable State feature, that mounts / dismounts every component. To work around, disable React Strict Mode.

I'm still looking into the "npm ERR! code ERESOLVE" issue.

**Comment by ag-grid-shared** [comment]
I can't reproduce "npm ERR! code ERESOLVE" issue.

I followed instructions here to get latest version:
https://create-react-app.dev/docs/getting-started/
i.e. I ran _"npm uninstall -g create-react-app"_ first.

Any help for me to recreate?

**Comment by juhahinkula** [comment]
> I can't reproduce "npm ERR! code ERESOLVE" issue.
> 
> I followed instructions here to get latest version: https://create-react-app.dev/docs/getting-started/ i.e. I ran _"npm uninstall -g create-react-app"_ first.
> 
> Any help for me to recreate?

Strange. I have tried both with Powershell and Ubuntu bash and I can't install ag-grid. Here is my dependencies from the package.json file and from there you can see that I am using the latest react-scripts version.

"dependencies": {
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.0.1",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },

**Comment by juhahinkula** [comment]
> My dependencies are identical. I'm on OSX and using the normal OSC terminal.
> 
> I don't have easy access to a Windows machine at the moment. I'll see if I can grab one. Can anyone else verify this???
> 
> When you say Powershell, is that using Windows binaries? And when you say Ubuntu bash, is that WSL 2?

Yes WSL2 and Powershell 7.2.2

**Comment by seanlandsman** [comment]
This is because ag-grid-react 27.1.0 has peer dependencies of 16 & 17 - React 18 wasn't available when 27.1.0 was released.

You can either wait for 27.2.0 which will be released next week (and has peer deps of 16, 17 and 18) or simply ignore the error (it'll work just fine) by adding the `--legacy-peer-deps` flag to `npm` - ie `npm install --legacy-peer-deps`

**Comment by juhahinkula** [comment]
> This is because ag-grid-react 27.1.0 has peer dependencies of 16 & 17 - React 18 wasn't available when 27.1.0 was released.
> 
> You can either wait for 27.2.0 which will be released next week (and has peer deps of 16, 17 and 18) or simply ignore the error (it'll work just fine) by adding the `--legacy-peer-deps` flag to `npm` - ie `npm install --legacy-peer-deps`

Yes, the reason is clear. Thanks for releasing new version soon.

**Comment by ag-grid-shared** [comment]
Great, thanks @seanlandsman . With regards the ResizeObserver.observe issue, I've code complete a fix, assuming it pasts our internal testing it will probably be in a patch in the next week. So both issues are on their way to been solved.

**Comment by seanlandsman** [comment]
Release 27.2.1 has been issued which should address the above issues

Thanks for reporting the issue!
